### PR TITLE
Signup: Clean up signup/stepper feature flags

### PR DESCRIPTION
**Pull request by arthur791004**
#### Proposed Changes

Clean up the following flags as it has been turned on for a long time

- [x] signup/design-picker-use-featured-picks-buttons
- [x] signup/ftm-flow-non-en
- [x] signup/starting-point-courses
- [x] signup/site-vertical-step
- [x] signup/hero-flow-non-en
- [x] signup/design-picker-premium-themes-checkout
- [x] signup/goals-step
- [x] signup/goals-step-2
- [x] signup/reskin
- [x] signup/seller-upgrade-modal
- [x] signup/anchor-fm
- [x] signup/design-picker-categories

#### Testing Instructions

**New site**

* Create a new site with the free plan
* Go through the `site-setup` flow
* Select free/premium themes
* Ensure you're able to finish the flow

**Anchor FM**

* Go to `/setup/?anchor_podcast=9370e048`
* Make sure you can complete the flow

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to https://github.com/Automattic/wp-calypso/issues/68244

### CODE REVIEW

File: client/components/theme/test/__snapshots__/index.jsx.snap [hunk 1]
```diff
@@ -30,9 +30,6 @@ exports[`Theme rendering with default display buttonContents should match snapsh
       >
         Twenty Seventeen
       </h2>
-      <span
-        class="theme__badge-price theme__badge-price-upgrade"
```
Comment (taipeicoder) [inline client/components/theme/test/__snapshots__/index.jsx.snap:34]: Could you explain why was this removed? I don't quite understand it 😅

**Comment by taipeicoder** [comment]
Tested:
- [x] New site with free plan and selecting a free theme.
- [x] New site with premium plan (upgraded from the plans step), and selecting a premium theme.
- [x] New site with free plan and skipping in the goals screen.
- [x] New site with free plan and skipping in the design picker screen.
- [x] New site with Anchor.fm link.

Nice clean up! ❤️

File: client/components/theme/test/__snapshots__/index.jsx.snap (hunk 1 above)
Comment (arthur791004) [reply client/components/theme/test/__snapshots__/index.jsx.snap:34]: Below is the original code before we remove the feature flag. However, the value is false because we don't turn on the flag in the test environment. Thus, the price will show in snapshots.
```js
{ ( ! isEnabled( 'signup/seller-upgrade-modal' ) || active ) && (
  <span className={ priceClass }>{ price }</span>
) }
```

Comment (taipeicoder) [reply client/components/theme/test/__snapshots__/index.jsx.snap:34]: Ah okay! I thought it would still show up since the test was changed:
```diff
-const { container } = render( <Theme { ...props } price="$50" /> );
+const { container } = render( <Theme { ...props } price="$50" active /> );
```

The price would still show up since:
```
{ active && <span className={ priceClass }>{ price }</span> }
```

Maybe I misunderstood, and the snapshot is not related to the test?

Comment (arthur791004) [reply client/components/theme/test/__snapshots__/index.jsx.snap:34]: The snapshot is generated from [here](https://github.com/Automattic/wp-calypso/blob/4260dcad4429914e261dad0610e16fcd028a937e/client/components/theme/test/index.jsx#L78)

```js
test( 'should match snapshot', () => {
  const { container } = render( <Theme { ...props } /> );
  expect( container.firstChild ).toMatchSnapshot();
} );
```

**Review by taipeicoder — APPROVED** [review]
